extensions/amp-story/1.0/amp-story.js

```diff
@@ -404,12 +404,19 @@ export class AmpStory extends AMP.BaseElement {
 
     if (isExperimentOn(this.win, 'amp-story-branching')) {
       this.registerAction('goToPage', invocation => {
-        const {args} = invocation;
+        const {args, caller} = invocation;
+
         if (args) {
           this.storeService_.dispatch(
               Action.SET_ADVANCEMENT_MODE, AdvancementMode.GO_TO_PAGE);
-          this.switchTo_(args['id'], NavigationDirection.NEXT);
-        }
+
+          if (caller['offsetParent'].tagName === 'AMP-SIDEBAR') {
+            this.sidebar_.getImpl()
+                .then(sidebarImpl => sidebarImpl.close_());
+            this.switchTo_(args['id'], NavigationDirection.NEXT);
+            this.closeOpacityMask_();
+          }
+        } this.switchTo_(args['id'], NavigationDirection.NEXT);
```

Comment:
nit: Can you put this on its own line, instead of on the same line as the brace?

Reply:
Done